Tiltfile

```diff
@@ -1,6 +1,9 @@
 # -*- mode: Python -*-
 
-load("ext://uibutton", "cmd_button", "text_input")
+envsubst_cmd = "./hack/tools/bin/envsubst"
+kubectl_cmd = "/usr/bin/kubectl"
```

Comment:
I don't think we should hard-code kubectl.
Can we only rely on it being in the path?

Reply:
Not sure, we can do it, but this requires an additional target which installs kubectl. 

As we don't depend on any specific kubectl version it might be enough to just rely on it being in the path and documenting it (I guess 99% of our users have kubectl)

Reply:
Sure I'll see if I can figure something out with Tilt's native features.

Reply:
@sbueringer So I looked into the docs for a [k8s resource](https://docs.tilt.dev/api.html#api.k8s_resource) and it looks like it expects to have a yaml or other static workload that it applies when the Tiltfile loads. In our case, if we want to use commands like envsubst, clusterctl, or getting a random # from the shell to dynamically configure the resource once Tilt is running, I think we'd need to handle applying it ourselves. I saw that there is a [k8s_custom_deploy](https://docs.tilt.dev/api.html#api.k8s_custom_deploy), but it expects you to give it a command to apply the resource using the shell, which is what we're trying to avoid.

Reply:
I wonder if we can do something similar to how we deploy helm charts today. We use tilt-prepare to render YAML files and then Tilt just picks them up and deploys them when necessary. But I guess this depends on the outcome of the clusterctl discussion below.